File: app/lib/models/grouprequest.dart
class Grouprequest {
  String? id;
  String? groupId;
  dynamic userId; 

  String? requestAt;
  

  Grouprequest(
      {this.id,
      this.groupId,
      this.userId,
      this.requestAt,
  });

  Grouprequest.fromJson(Map<String, dynamic> json) {
    id = json['_id'];
    groupId = json['groupId'];
    userId = json['userId'];
    requestAt = json['requestAt'];
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'groupId': groupId,
      'userId': userId,
      'requestAt': requestAt,
    };
  }
}

File: api/models/GroupRequest.js
const mongoose = require('mongoose');

const groupRequestSchema = new mongoose.Schema({
    groupId: {
        type: mongoose.Schema.Types.ObjectId,
        required: true,
        ref: 'Group'
    },
    userId: {
        type: mongoose.Schema.Types.ObjectId,
        required: true,
        ref: 'User'
    },
    requestAt: {
        type: Date,
        default: Date.now
    },
}, { timestamps: true });

module.exports = mongoose.model('GroupRequest', groupRequestSchema);